### PR TITLE
Add a `Bearer` security schema support

### LINKED ISSUE #1096: Refinement of security section using OpenAPI 3
Please check if the PR https://github.com/zalando/restful-api-guidelines/pull/551 introduces changes which are relevant to the Zally project.

### PR DESCRIPTION
**Pull request by vadeg**
* Add a `Bearer` security schema support
* Clean up the code 

Fixes #1096

### CODE REVIEW

File: server/zally-ruleset-zalando/src/main/kotlin/org/zalando/zally/ruleset/zalando/SecureAllEndpointsWithScopesRule.kt [hunk 1]
```diff
@@ -46,56 +53,76 @@ class SecureAllEndpointsWithScopesRule(rulesConfig: Config) {
...
+                                when {
+                                    schema.isOAuth2() -> {
+                                        validateOAuth2Schema(
+                                            context,
+                                            op,
+                                            operationScopes,
+                                            schema,
+                                            schemaName
+                                        )
+                                    }
```
Comment (tkrop) [inline server/zally-ruleset-zalando/src/main/kotlin/org/zalando/zally/ruleset/zalando/SecureAllEndpointsWithScopesRule.kt:81]: The brackets are not necessary - which may improve readability here.

**Comment by tkrop** [comment]
:+1:

**Comment by tkrop** [comment]
May be we need to investigate why the build failed before merging:
```
=== RUN   TestIntegrationWithNoMustViolations
--- FAIL: TestIntegrationWithNoMustViolations (0.33s)
    integration_test.go:84: 
        	Error Trace:	integration_test.go:84
        	Error:      	Should be zero, but was 9
        	Test:       	TestIntegrationWithNoMustViolations
        	Messages:   	No MUST violations expected
    integration_test.go:88: 
        	Error Trace:	integration_test.go:88
        	Error:      	Expected nil, but got: &domain.AppError{e:(*errors.errorString)(0xc000167050), c:3}
        	Test:       	TestIntegrationWithNoMustViolations
`

**Review by tkrop — APPROVED** [review]

**Comment by vadeg** [comment]
👍

**Comment by tkrop** [comment]
:+1:

**Pull request merged**